src/renate/cli/parsing_functions.py

```diff
@@ -134,10 +135,35 @@ def get_updater_and_learner_kwargs(
     return updater_class, learner_kwargs
 
 
+def _cast_arguments(arguments: Dict[str, Dict[str, Any]], args_dict: Dict[str, str]) -> None:
```

Comment:
Naming of the function arguments is confusing, and so is the doc string. Can we improve that?